src/main/resources/conf/configuration.xsd

```diff
@@ -0,0 +1,371 @@
+        the other for exclusion,
+      </xs:documentation>
+    </xs:annotation>
+    <xs:sequence>
+      <xs:element maxOccurs="unbounded" minOccurs="0"
+                  name="exclude" type="xs:normalizedString"/>
+      <xs:element maxOccurs="unbounded" minOccurs="0"
+                  name="include" type="xs:normalizedString"/>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="harvest_configuration_type">
+    <xs:annotation>
+      <xs:documentation xml:lang="en">
+        These are the basic options for the harvest configuration file.
+
+        @nodeName: the PDS node that this harvest run applies to
+
+        autogenFields: should not be used except in development testing
+        do: tells where and how to harvest PDS4 labels
+        fileInfo: option allowing filename prefixes to be replaced
+        productFilter: should not be used except in development testing
+        registry: define the server harvest should use
+        xpathMaps: allow constraints in the PDS4 label to control harvesting
+      </xs:documentation>
+    </xs:annotation>
+    <xs:all>
+      <xs:element minOccurs="0"
+                  name="autogenFields" type="autogen_fields_type"/>
+      <xs:element name="do" type="do_type"/>
+      <xs:element minOccurs="0" name="fileInfo" type="file_info_type"/>
+      <xs:element minOccurs="0" name="productFilter" type="filter_type"/>
+      <xs:element minOccurs="0" name="references" type="references_type"/>
+      <xs:element name="registry" type="registry_type"/>
+      <xs:element minOccurs="0" name="xpathMaps" type="xpath_maps_type"/>
+    </xs:all>
+    <xs:attribute name="nodeName" type="node_name_enum" use="required"/>
+  </xs:complexType>
+
+  <xs:complexType name="product_type">
+    <xs:annotation>
+      <xs:documentation xml:lang="en">
+      </xs:documentation>
+    </xs:annotation>
+    <xs:attribute name="dir" type="xs:normalizedString" use="required"/>
+  </xs:complexType>
+
+  <xs:complexType name="references_type">
+    <xs:annotation>
+      <xs:documentation xml:lang="en">
+        Seems like a forgotten idea
+      </xs:documentation>
+    </xs:annotation>
+    <xs:attribute default="false" name="primaryOnly" type="xs:boolean"/>
+  </xs:complexType>
+
+  <xs:complexType name="registry_type">
+    <xs:annotation>
+      <xs:documentation xml:lang="en">
+        Define the connection to the registry, security for the connection, and
+        the index within the registry.
+
+        @auth: a java property file containing a username and password
+        @index: the index to be used by harvest whose default is registry
+        @trust_self_signed: all self signed certificates for https
+
+        client_id: the cognito client ID for AWS based instances of opensearch
+        direct_url: the opensearch URL when not using AWS services
+      </xs:documentation>
+    </xs:annotation>
+    <xs:choice>
+      <xs:element name="client_id" type="xs:normalizedString"/>
```

Comment:
@al-niessner can we rename this `cognito_client_id`? I know it is wordy, but I would like it to be self-documenting in a why in the config.